Add Intermediate steps docs (#260)
agno-agi/agno-docs: agents/run.mdx

| `response_audio` | `ModelResponseAudio`   | `None`                        | The model's raw response in audio.                                                                                               |
| `created_at`     | `int`                  | -                             | Unix timestamp of the response creation.                                                                                         |
| `extra_data`     | `RunResponseExtraData` | `None`                        | Extra data containing optional fields like `references`, `add_messages`, `history`, `reasoning_steps`, and `reasoning_messages`. |

## Streaming Intermediate Steps

> You can enable streaming of intermediate steps by setting `stream_intermediate_steps=True`.

Author: But ok with your suggestion too

```python
# Stream with intermediate steps
response_stream = agent.run(
    "Tell me a 5 second short story about a lion",
    stream=True,
    stream_intermediate_steps=True
)
```

### Event Types

> When streaming intermediate steps, Events that can be sent by the `Agent.run()` function:

Reviewer: ```suggestion
The following events are sent by the `Agent.run()` and `Agent.arun()` functions:
```

Author: But not all events are sent `are sent` says they all will be sent. their actual occurrence depends on the agent's configuration. 

Should we write like:

The following events are sent by the `Agent.run()` and `Agent.arun()` functions depending on agent's configuration:

Reviewer: Yes happy with that

| Event Type | Description |
|------------|-------------|
| `RunStarted` | Indicates the start of a run |

> | `RunResponse` | Contains the model's response |

Reviewer: ```suggestion
| `RunResponse` | Contains the model's response text |
```
